### PR TITLE
pythonPackages.ratelim: init at 0.1.6

### PR DESCRIPTION
**Pull request by dgliwka**
###### Motivation for this change

Simple Python library that limits the number of times a function can be called during a time interval

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129843` at b9931849 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.ratelim</li>
    <li>python39Packages.ratelim</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/ratelim/default.nix:26:0:
```
   |
26 |     description = "Simple Python library that limits the number of times a function can be called during a time interval";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/ratelim/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    repo = pname;
+    rev = "master";
+    sha256 = "0qznfp5jayp612yv7zrxz1mfp34n1zqa6aj8i1zjrrhiv3zfq7fx";
+  };
+
+  propagatedBuildInputs = [
+    decorator
+  ];
+
+  pythonImportsCheck = [ "ratelim" ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/ratelim/default.nix:22]: Disable the checkPhase if the library doesn't have any tests.

```suggestion
  pythonImportsCheck = [ "ratelim" ];
  
  # package has no tests
  doCheck = false;
```

But we could run https://github.com/themiurgo/ratelim/blob/0.1.1/ratelim/__init__.py#L71-L90 instead?

**Review by mweinelt — CHANGES_REQUESTED** [review]

**Comment by mweinelt** [comment]
Thanks for contributing to nixpkgs. This looks pretty good already. I added one change request regarding tests.

**Review by fabaff — COMMENTED** [review]
This module doesn't seems to be under active development or maintenance for quite some time. If it's not a dependency of another packages then I think that we should stick with the limiting modules which we already have.

File: pkgs/development/python-modules/ratelim/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, decorator
+}:
+
+buildPythonPackage rec {
+  pname = "ratelim";
+  version = "0.1.1";
```
[the remark this answers is not in the document]
Comment (dgliwka) [reply pkgs/development/python-modules/ratelim/default.nix]: It is a dependency for [Geocoder](https://geocoder.readthedocs.io) – library I would like to use.
I modified package code to get version 0.1.6 from PyPI.

**Pull request merged**